tests/lib/journalctl.sh

```diff
@@ -1,23 +1,58 @@
 #!/bin/bash
 
-JOURNALCTL_CURSOR_FILE="${SPREAD_PATH}"/journalctl_cursor
+# shellcheck source=tests/lib/state.sh
+. "$TESTSLIB/state.sh"
+
+# This file contains all the cursors used for the different tests
+JOURNALCTL_CURSOR_FILE="$RUNTIME_STATE_PATH"/journalctl_cursor
 
 get_last_journalctl_cursor(){
     journalctl --output=export -n1 | grep --binary-files=text -o '__CURSOR=.*' | sed -e 's/^__CURSOR=//'
 }
 
+restart_journalctl(){
+    systemctl restart systemd-journald
+}
+
 start_new_journalctl_log(){
+    restart_journalctl
+    echo "New test starts here - $SPREAD_JOB" | systemd-cat -t snapd-test
     cursor=$(get_last_journalctl_cursor)
     if [ -z "$cursor" ]; then
         echo "Empty journalctl cursor, exiting..."
         exit 1
     else
-        echo "$cursor" > "$JOURNALCTL_CURSOR_FILE"
+        echo "$SPREAD_JOB " >> "$JOURNALCTL_CURSOR_FILE"
+        echo "$cursor" >> "$JOURNALCTL_CURSOR_FILE"
     fi
 }
 
+check_journalctl_ready(){
+    marker="test-${RANDOM}${RANDOM}"
+    echo "Running test: $marker" | systemd-cat -t snapd-test
+    if check_journalctl_log "$marker"; then
+        return 0
+    fi
+    echo "Test id not found in journalctl, exiting..."
+    exit 1
+}
+
+check_journalctl_log(){
+    expression=$1
+    shift
+    for _ in $(seq 20); do
+        log=$(get_journalctl_log "$@")
+        if echo "$log" | grep -q -E "$expression"; then
```

Comment:
Still feels like we're doing something wrong and maybe need `--sync` at some random places. I would expect that if a message is logged it ends up in the journal in either runtime or backing store locations. Subsequent queries should return the logged message from either place. Maybe journald does not give such guarantees and the fix that we do here just helps us achieve eventual consistency state.

Reply:
@bboozzoo the problema around --sync is the it was getting stuck and the test suite was failing because of that. Also --sync is not available in the version we are using on ubuntu 14.04. 
I think it is a delay until the journal gets a message and another delay until its returns that message as part of the log. 

There is a log where we can see how the debug info from a snap is diplayed after 2 seconds it was generated, even after we remove that snap.